utils/testman/tests/changeset_test.go

```diff
@@ -0,0 +1,178 @@
+package tests
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/acorn-io/manager/tests/e2e"
+	"github.com/acorn-io/manager/tests/helper"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func randToken(length int) (string, error) {
+	randBytes := make([]byte, length)
+
+	_, err := rand.Read(randBytes)
+	if err != nil {
+		return "", err
+	}
+
+	return hex.EncodeToString(randBytes), nil
+}
+
+func trimToDir(path, targetDir string) (string, error) {
+	// Split the path into its components
+	components := strings.Split(path, string(filepath.Separator))
+
+	// Find the index of the target directory
+	targetIndex := -1
+	for i, component := range components {
+		if component == targetDir {
+			targetIndex = i
+			break
+		}
+	}
+
+	// If the target directory is not found, return an error
+	if targetIndex == -1 {
+		return "", fmt.Errorf("target directory '%s' not found in path", targetDir)
+	}
+
+	// Construct the new path from the beginning to the target directory
+	newPath := filepath.Join(components[:targetIndex+1]...)
+	return "/" + newPath, nil
```

Comment:
Why the leading `/` here? Maybe this is just a `filepath.Abs`?

Reply:
The leading slash is because the trimmed path was already absolute but `filepath.Join` doesn't account for that. `filepath.Abs` doesn't work because it needs a relative path not an absolute one.

Reply:
You might want to use the `filepath.Separator` value or whatever the exact one is so that this can run without issue on different platforms

Reply:
Good point. Fixed that. :+1: